adapter/src/components/AuthBoundary.js

```diff
@@ -0,0 +1,112 @@
+import {
+    useConfig,
+    useDataQuery,
+    clearSensitiveCaches,
+} from '@dhis2/app-runtime'
+import { CenteredContent, NoticeBox } from '@dhis2/ui'
+import PropTypes from 'prop-types'
+import React, { useState } from 'react'
+import i18n from '../locales'
+import { LoadingMask } from './LoadingMask'
+import styles from './styles/ErrorBoundary.style.js'
+
+const LATEST_USER_KEY = 'dhis2.latestUser'
+const IS_PRODUCTION_ENV = process.env.NODE_ENV === 'production'
+
+const APP_MANAGER_AUTHORITY = 'M_dhis-web-maintenance-appmanager'
+const REQUIRED_APP_AUTHORITY = process.env.REACT_APP_DHIS2_APP_AUTH_NAME
+
+const USER_QUERY = {
+    user: {
+        resource: 'me',
+        params: { fields: ['id', 'authorities'] },
+    },
+}
+
+async function clearCachesIfUserChanged({ currentUserId, pwaEnabled }) {
+    const latestUserId = localStorage.getItem(LATEST_USER_KEY)
+    if (currentUserId !== latestUserId) {
+        const cachesCleared = await clearSensitiveCaches()
+        localStorage.setItem(LATEST_USER_KEY, currentUserId)
+        if (cachesCleared && pwaEnabled) {
+            // If this is a PWA app, the app-shell cache will need to
+            // be restored with a page reload
+            return window.location.reload()
+        }
+    }
+}
+
+const isAppAvailable = (authorities) => {
+    // Skip check on dev
+    // TODO: should we check on dev environments too?
+    if (!IS_PRODUCTION_ENV) {
+        return true
+    }
```

Comment:
Curious what others think -- most app devs probably have admin privileges, but if they don't have app management authority, it's possible that someone might end up working on an a new app which they don't have authorities for.

Someone might be able to bypass their authorities limitations by running an app locally though?...

Currently I guess we don't do any checks like this

Reply:
(You can make testing easier by commenting out the `return true` to run it on dev)

Reply:
I initially wasn't sold on the idea of always returning true in a development environment, but think I'm coming around to it.
|         | Full reload     | From SW |
|--------------|-----------|------------|
| Production | Server renders no-access page      | App-shell renders no-access view        |
| Development      | Dev can access app  | Dev can access app       |